dlib/test/fft.cpp

```diff
@@ -171,6 +359,14 @@ namespace
             test_against_saved_good_ffts();
             test_random_ffts();
             test_random_real_ffts();
+            test_linearity_real<float>();
+            test_linearity_real<double>();
+            test_linearity_complex<float>();
+            test_linearity_complex<double>();
+            test_kronecker_delta_impulse_response<float>();
+            test_kronecker_delta_impulse_response<double>();
+            test_time_shift<float>();
+            test_time_shift<double>();
         }
```

Comment:
Be sure to test FFTs that have sizes that are not even.  Like check ffts of size 1, 3, 123, and other oddball sizes since this new stuff is supposed to support ffts of any size.

Reply:
Yep. I think in general there should be more unit tests. Like the DFTs of standard window functions could be computed and checked analytically. And then add for-loops around them for different FFT sizes.